build/tidy.mk

```diff
@@ -30,5 +30,5 @@ Clang_Tidy_CXX_Files = $(filter %.cc, $(sort $(1)))
 #
 # All this clearly requires GNU make.
 
-CXX_Clang_Tidy = $(foreach tidy_target, $(call Clang_Tidy_CXX_Files,$^), $(CLANG_TIDY) $(Clang_Tidy_Options) $(tidy_target) -- $(CXXCOMPILE) -x c++;)
```

Comment:
Didn't see wired changes like `std::move(std::move(std::move(std::move(...` ?
/cc #5172

Reply:
No, I didn't see any of those types of weird changes.  It wasn't able to run in the plugins directory prior to this change because the executable string was too large.

Reply:
It looks like modernize-use-default-member-init is failing on all the files for some reason in Fedora 30.  For now we should disable this feature and we can add it back once it becomes more stable.  Here is the coredump:

```
08:53:23 homer:(clang_tidy_updates_9.0)~/dev/apache/trafficserver$ make clang-tidy
Making clang-tidy in src/tscpp/util
make[1]: Entering directory '/home/bcall/dev/apache/trafficserver/src/tscpp/util'
/usr/bin/clang-tidy -fix -fix-errors TextView.cc unit_tests/test_MemSpan.cc unit_tests/test_PostScript.cc unit_tests/test_TextView.cc unit_tests/test_ts_meta.cc unit_tests/unit_test_main.cc -- /usr/bin/clang++ -DHAVE_CONFIG_H -I. -I../../../include  -Dlinux -D_LARGEFILE64_SOURCE=1 -D_COMPILE64BIT_SOURCE=1 -D_REENTRANT -D__STDC_LIMIT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -I/home/bcall/dev/apache/trafficserver/include -D_GNU_SOURCE -DOPENSSL_NO_SSL_INTERN -std=c++17 -g -pipe -Wall -Wno-deprecated-declarations -Qunused-arguments -Wextra -Wno-ignored-qualifiers -Wno-unused-parameter -O3 -fno-strict-aliasing -Werror -Wno-invalid-offsetof -mcx16 -stdlib=libc++ -I/usr/include/c++/v1 -I/usr/include/luajit-2.1/ -std=c++17 -x c++
 #0 0x00007f98398f706e llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/lib64/libLLVM-8.so+0xabd06e)
 #1 0x00007f98398f5014 llvm::sys::RunSignalHandlers() (/lib64/libLLVM-8.so+0xabb014)
 #2 0x00007f98398f5195 (/lib64/libLLVM-8.so+0xabb195)
 #3 0x00007f98374cff40 __restore_rt (/lib64/libc.so.6+0x37f40)
 #4 0x00007f9837d03201 llvm::ManagedStaticBase::RegisterManagedStatic(void* (*)(), void (*)(void*)) const (/lib64/libclangTidyModernizeModule.so.8+0xc3201)
 #5 0x00007f9837dfbd59 clang::tidy::modernize::UseDefaultMemberInitCheck::checkDefaultInit(clang::ast_matchers::MatchFinder::MatchResult const&, clang::CXXCtorInitializer const*) (/lib64/libclangTidyModernizeModule.so.8+0x1bbd59)
 #6 0x00007f9837dfc2b8 clang::tidy::modernize::UseDefaultMemberInitCheck::check(clang::ast_matchers::MatchFinder::MatchResult const&) (/lib64/libclangTidyModernizeModule.so.8+0x1bc2b8)
 #7 0x00007f9838926958 (/lib64/libclangASTMatchers.so.8+0x36958)
 #8 0x00007f983896ad80 clang::ast_matchers::internal::BoundNodesTreeBuilder::visitMatches(clang::ast_matchers::internal::BoundNodesTreeBuilder::Visitor*) (/lib64/libclangASTMatchers.so.8+0x7ad80)
 #9 0x00007f98389287b4 (/lib64/libclangASTMatchers.so.8+0x387b4)
#10 0x00007f983894e912 (/lib64/libclangASTMatchers.so.8+0x5e912)
#11 0x00007f9838951366 (/lib64/libclangASTMatchers.so.8+0x61366)
#12 0x00007f983894fbcf (/lib64/libclangASTMatchers.so.8+0x5fbcf)
#13 0x00007f9838951366 (/lib64/libclangASTMatchers.so.8+0x61366)
#14 0x00007f983894ea8c (/lib64/libclangASTMatchers.so.8+0x5ea8c)
#15 0x00007f9838951366 (/lib64/libclangASTMatchers.so.8+0x61366)
#16 0x00007f983894ea8c (/lib64/libclangASTMatchers.so.8+0x5ea8c)
#17 0x00007f9838951366 (/lib64/libclangASTMatchers.so.8+0x61366)
#18 0x00007f983894eaac (/lib64/libclangASTMatchers.so.8+0x5eaac)
#19 0x00007f98389509f3 clang::ast_matchers::MatchFinder::matchAST(clang::ASTContext&) (/lib64/libclangASTMatchers.so.8+0x609f3)
#20 0x00007f98370adce0 clang::MultiplexConsumer::HandleTranslationUnit(clang::ASTContext&) (/lib64/libclangFrontend.so.8+0x129ce0)
#21 0x00007f98358a3469 clang::ParseAST(clang::Sema&, bool, bool) (/lib64/libclangParse.so.8+0x3c469)
#22 0x00007f9837073066 clang::FrontendAction::Execute() (/lib64/libclangFrontend.so.8+0xef066)
#23 0x00007f9837025406 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/lib64/libclangFrontend.so.8+0xa1406)
#24 0x00007f983792de6f clang::tooling::FrontendActionFactory::runInvocation(std::shared_ptr<clang::CompilerInvocation>, clang::FileManager*, std::shared_ptr<clang::PCHContainerOperations>, clang::DiagnosticConsumer*) (/lib64/libclangTooling.so.8+0x72e6f)
#25 0x00007f98385607f8 (/lib64/libclangTidy.so.8+0x3a7f8)
#26 0x00007f9837924993 clang::tooling::ToolInvocation::runInvocation(char const*, clang::driver::Compilation*, std::shared_ptr<clang::CompilerInvocation>, std::shared_ptr<clang::PCHContainerOperations>) (/lib64/libclangTooling.so.8+0x69993)
#27 0x00007f98379297ee clang::tooling::ToolInvocation::run() (/lib64/libclangTooling.so.8+0x6e7ee)
#28 0x00007f983792c2e6 clang::tooling::ClangTool::run(clang::tooling::ToolAction*) (/lib64/libclangTooling.so.8+0x712e6)
#29 0x00007f9838574a73 clang::tidy::runClangTidy(clang::tidy::ClangTidyContext&, clang::tooling::CompilationDatabase const&, llvm::ArrayRef<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>, bool, llvm::StringRef) (/lib64/libclangTidy.so.8+0x4ea73)
#30 0x0000564721dd7f49 (/usr/bin/clang-tidy+0x32f49)
#31 0x00007f98374bbf33 __libc_start_main (/lib64/libc.so.6+0x23f33)
#32 0x0000564721dc81ee LLVMInitializeXCoreTargetMC (/usr/bin/clang-tidy+0x231ee)
make[1]: *** [Makefile:1431: clang-tidy-local] Segmentation fault (core dumped)
make[1]: Leaving directory '/home/bcall/dev/apache/trafficserver/src/tscpp/util'
make: *** [Makefile:850: clang-tidy-recursive] Error 1
```